contentcuration/contentcuration/utils/publish.py

```diff
@@ -226,15 +227,19 @@ def create_bare_contentnode(ccnode, default_language, channel_id, channel_name):
     if ccnode.language or default_language:
         language, _new = get_or_create_language(ccnode.language or default_language)
 
-    duration = None
-    if ccnode.kind_id in [content_kinds.AUDIO, content_kinds.VIDEO]:
-        # aggregate duration from associated files, choosing maximum if there are multiple, like hi and lo res videos
-        duration = ccnode.files.aggregate(duration=Max("duration")).get("duration")
-
     options = {}
     if ccnode.extra_fields and 'options' in ccnode.extra_fields:
         options = ccnode.extra_fields['options']
 
+    duration = None
+    ccnode_completion_criteria = options.get("completion_criteria")
+    if ccnode_completion_criteria:
+        if ccnode_completion_criteria["model"] == completion_criteria.TIME or ccnode_completion_criteria["model"] == completion_criteria.APPROX_TIME:
+            duration = ccnode_completion_criteria["threshold"]
+    elif ccnode.kind_id in [content_kinds.AUDIO, content_kinds.VIDEO]:
```

Comment:
This might be better as `if duration is None and ...`, but I don't think it will have any effect on Kolibri